app/main.py

```diff
@@ -1,8 +1,32 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_dict in people:
+        name = person_dict["name"]
+        age = person_dict["age"]
+        person = Person(name, age)
+        person_list.append(person)
+
+    for person_dict in people:
+        name = person_dict["name"]
+        spouse_name = person_dict.get("wife") or person_dict.get("husband")
```

Comment:
The use of `get` with `or` to determine `spouse_name` is correct, but it assumes that the dictionary will always have either 'wife' or 'husband'. Ensure that this assumption holds true for all inputs to avoid unexpected behavior.